### PR TITLE
Add pod monitor configs for envoy cluster

### PR DESCRIPTION
**Pull request by gammazero**

### CODE REVIEW

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/pod-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  selector:
+    matchLabels:
+      app: envoy
+  namespaceSelector:
+    matchNames:
+      - storetheindex
+  podMetricsEndpoints:
+    - path: /metrics
+      port: metrics
```
Comment (masih) [inline deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/pod-monitor.yaml:16]: envoy port is called `admin` i think

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/pod-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    app: envoy
+spec:
+  selector:
+    matchLabels:
+      app: envoy
+  namespaceSelector:
+    matchNames:
+      - storetheindex
+  podMetricsEndpoints:
+    - path: /metrics
```
Comment (masih) [inline deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/pod-monitor.yaml:15]: The path at which metrics are exposed in envoy is also different. This value should be the same as the one in `ServiceMonitor`

Comment (gammazero) [reply deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/pod-monitor.yaml:15]: I did copy this from [service-monitor.yaml](https://github.com/ipni/storetheindex/blob/main/deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/instances/service-monitor.yaml), I think you mean I need to use what is in [envoy-monitor.yaml](https://github.com/ipni/storetheindex/blob/main/deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/envoy-monitor.yaml)

Comment (gammazero) [reply deploy/manifests/dev/us-east-2/tenant/storetheindex/dhstore-cluster/envoy/pod-monitor.yaml]: Done

**Review by masih — COMMENTED** [review]
spec looks right thanks Andrew 👍 They also need to be added to their `kustomization.yaml` under `resources` to be included in the generation.

**Review by masih — APPROVED** [review]

**Comment by gammazero** [comment]
Not using dhstore cluster

**Pull request closed without merging**